tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-navigation-access.bal

```diff
@@ -170,6 +170,25 @@ function testXMLNavigationWithEscapeCharacter() {
     assert(x7, xml `<home-address>some address</home-address>`);
 }
 
+function testXmlNavigationWithDefaultNamespaceDefinedLater() {
+    xml x1 = xml `<a>a<b>b<c>c</c></b><d>d</d></a><f/>`;
+
+    assert((x1.<a>).toString(), "<a>a<b>b<c>c</c></b><d>d</d></a>");
```

Comment:
Using `toString()` may not be the best way to test this.
Can't we directly assert the xml values.

Reply:
We can't use the direct `xml` values to compare here, because the xml values we use for comparison is effected by the default xml namespace defined. For example the in line [186](https://github.com/poorna2152/ballerina-lang/blob/xml_nav_stmt_2/tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-navigation-access.bal#L186) the `xml` value we use for assert has the default namespace defined before that as the namespace of that value. But this namespace is not present in the value we get from the navigation. Thus we have to use `toString` here.

Reply:
The problem is with this approach, the type will be not checked.
But since we have other test cases it will not be a problem here.
+1

Reply:
Can't we move the assertion to a separate function? Will make it a bit lengthy, but will also make it a more accurate test IMO.

Reply:
Updated